idaes/generic_models/unit_models/surrogate_model.py

```diff
@@ -0,0 +1,135 @@
+    """
+
+    # Create Class ConfigBlock
+    CONFIG = ProcessBlockData.CONFIG()
+    CONFIG.declare("dynamic", ConfigValue(
+        default=False,
+        domain=In([False]),
+        description="Dynamic model flag",
+        doc="""Surrogate model assumed to be steady-state."""))
+
+    def build(self):
+        """
+        General build method for SurrogateModelBlockData.
+
+        Inheriting models should call `super().build`.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        super(SurrogateModelData, self).build()
+
+    def add_ports(self, name=None, member_list=None, doc=None):
+        """
+        This is a method to build Port objects in a surrogate model and
+        populate this with appropriate port members as specified. User can add
+        as many inlet and outlet ports as required.
+
+        Keyword Args:
+            name : name to use for Port object.
+            dict : dictionary containing variables to be added to the port
+            doc : doc string for Port object
+
+        Returns:
+            A Pyomo Port object and associated components.
+        """
+        # Validate that member_list is a dict
+        if not isinstance(member_list, dict):
+            raise ConfigurationError(
+                "member_list should be a dictionary "
+                "with the keys being the name assigned(strings) and "
+                "values being the variable objects declared in the model.")
+
+        # Create empty Port
+        p = Port(noruleinit=True, doc=doc)
+        setattr(self, name, p)
+
+        # Create References for port members
+        for k in member_list.keys():
+            if not member_list[k].is_indexed():
+                local_name = member_list[k]
+            else:
+                local_name = member_list[k]
+
+            # Add Reference to Port
+            p.add(Reference(local_name), k)
+
+    def initialize(self, custom_initialize=None, outlvl=idaeslog.NOTSET,
+                   solver='ipopt', optarg={'tol': 1e-6}):
+        '''
+        This is a simple initialization routine for surrogate
+        models. If the user does not provide a custom callback function, this
+        method will check for degrees of freedom and if zero, will attempt
+        to solve the model as is.
+
+        Keyword Arguments:
+            custom_initialize : user callback for intialization
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating which solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        '''
+        # Set solver options
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = SolverFactory(solver)
+        opt.options = optarg
```

Comment:
I think you can replace these two lines with

```python
if optarg is None:
    optarg = {}
opt = get_solver(solver, oparg)
```

get_solver can be imported from idaes.core.util.misc

Reply:
`get_solver` now accepts `optarg=None` as an argument, so this can be done a a single line:

```
opt = get_solver(solver, optarg)
```